src/main/java/edu/ie3/datamodel/io/source/csv/CsvSystemParticipantSource.java

```diff
@@ -0,0 +1,594 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.source.csv;
+
+import edu.ie3.datamodel.io.FileNamingStrategy;
+import edu.ie3.datamodel.io.factory.EntityFactory;
+import edu.ie3.datamodel.io.factory.input.NodeAssetInputEntityData;
+import edu.ie3.datamodel.io.factory.input.participant.*;
+import edu.ie3.datamodel.io.source.RawGridSource;
+import edu.ie3.datamodel.io.source.SystemParticipantSource;
+import edu.ie3.datamodel.io.source.ThermalSource;
+import edu.ie3.datamodel.io.source.TypeSource;
+import edu.ie3.datamodel.models.UniqueEntity;
+import edu.ie3.datamodel.models.input.*;
+import edu.ie3.datamodel.models.input.container.SystemParticipants;
+import edu.ie3.datamodel.models.input.system.*;
+import edu.ie3.datamodel.models.input.system.type.*;
+import edu.ie3.datamodel.models.input.thermal.ThermalBusInput;
+import edu.ie3.datamodel.models.input.thermal.ThermalStorageInput;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.NotImplementedException;
+
+/**
+ * //ToDo: Class Description
+ *
+ * <p>TODO description needs hint that Set does NOT mean uuid uniqueness -> using the () getter //
+ * todo performance improvements in all sources to make as as less possible recursive stream calls
+ * on files without providing files with unique entities might cause confusing results if duplicate
+ * uuids exist on a file specific level (e.g. for types!)
```

Comment:
I think we have to harmonize the different class documentations.

Reply:
done